### PR TITLE
KOGITO-2154: Unit test cases for Management Console

### PR DESCRIPTION
**Pull request by Sara4994**
JIRA issue : https://issues.redhat.com/browse/KOGITO-2154

Description: Added unit test cases for react components on Management console

### CODE REVIEW

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/tests/DomainExplorerColumnPicker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+      enableCache: false,
+      setEnableCache: jest.fn(),
+      pageSize: 2,
+      offsetVal: 10,
+      setOffsetVal: jest.fn(),
+      setPageSize: jest.fn(),
+      setIsLoadingMore: jest.fn(),
+      isLoadingMore: true
+    };
+    // const paramFields = []
```
Comment (nmirasch) [inline packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/tests/DomainExplorerColumnPicker.test.tsx]: @Sara4994 remove this comment?

File: packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx [hunk 2]
```diff
@@ -190,6 +190,7 @@ const DomainExplorerDashboard = props => {
     return <ServerErrors error={getQuery.error} />;
   }
 
+  /* istanbul ignore if */
```
Comment (nmirasch) [inline packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]: @Sara4994  I'm just curious, this remove that ServerErrors component from coverage?

**Comment by nmirasch** [comment]
@Sara4994 Test looks good to me, I just found a bit weird that .snap needed update regarding the random generated keys. I'm not sure if that is something usual on this kind of testing.

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx [hunk 3]
```diff
@@ -221,17 +223,18 @@ const DomainExplorerColumnPicker: React.FC<IOwnProps> = ({
...
           .query({
             query: gql`
               ${Query.query}
             `,
             variables: Query.variables,
             fetchPolicy: enableCache ? 'cache-first' : 'network-only'
           })
-          .then(response => {
+          
+          // .then(response => {
```
Comment (nmirasch) [inline packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: @Sara4994 remove commented code?

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx [hunk 4]
```diff
@@ -252,7 +255,7 @@ const DomainExplorerColumnPicker: React.FC<IOwnProps> = ({
               setDisplayEmptyState(true);
               setEnableCache(false);
             }
-          });
+          // });
```
Comment (nmirasch) [inline packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: @Sara4994 same here :)

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx [hunk 5]
```diff
@@ -221,17 +223,18 @@ const DomainExplorerColumnPicker: React.FC<IOwnProps> = ({
           }
         }
       });
-
+      console.log(offsetVal, pageSize)
```
Comment (nmirasch) [inline packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: @Sara4994 is needed keep this log as part of testing?

Comment (Sara4994) [reply packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: no, its not required. removed it now :+1:

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx (hunk 4 above)
Comment (Sara4994) [reply packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: @nmirasch done

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx (hunk 3 above)
Comment (Sara4994) [reply packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx]: done

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/tests/DomainExplorerColumnPicker.test.tsx (hunk 1 above)
Comment (Sara4994) [reply packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/tests/DomainExplorerColumnPicker.test.tsx]: done

File: packages/management-console/package.json [hunk 6]
```diff
@@ -36,6 +36,7 @@
   },
   "dependencies": {
     "@kogito-apps/common": "1.0.0",
+    "@testing-library/react-hooks": "^3.2.1",
```
Comment (cristianonicolai) [inline packages/management-console/package.json]: dupliucated?

File: packages/management-console/package.json [hunk 7]
```diff
@@ -76,5 +78,9 @@
       "yarn run format",
       "yarn run lint"
     ]
+  },
+  "devDependencies": {
+    "@testing-library/react-hooks": "^3.2.1",
```
Comment (cristianonicolai) [inline packages/management-console/package.json]:
```suggestion
    "@testing-library/react-hooks": "3.2.1",
```

File: packages/management-console/package.json [hunk 8]
```diff
@@ -76,5 +78,9 @@
...
+    "core-js": "^3.6.5"
```
Comment (cristianonicolai) [inline packages/management-console/package.json]:
```suggestion
    "core-js": "3.6.5"
```

File: packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx (hunk 2 above)
Comment (cristianonicolai) [reply packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]: looks like it ignores the if below, not sure that something we want to skip intentionally?

Comment (Sara4994) [reply packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]: @nmirasch @cristianonicolai this one added by mistake, which is removed now and that block of code is covered in tests already

File: packages/management-console/package.json (hunk 6 above)
Comment (Sara4994) [reply packages/management-console/package.json]: @cristianonicolai actually we dont need that dependencies, removed it

File: packages/management-console/package.json (hunk 7 above)
Comment (Sara4994) [reply packages/management-console/package.json]: Removed the dependency itself

File: packages/management-console/package.json (hunk 8 above)
Comment (Sara4994) [reply packages/management-console/package.json]: @cristianonicolai caret removed

**Review by cristianonicolai — APPROVED** [review]

**Review by nmirasch — APPROVED** [review]

**Review by AjayJagan — APPROVED** [review]

**Pull request merged**